addons/website_sale/static/src/js/website_sale.editor.js

```diff
@@ -392,4 +446,62 @@ options.registry.ProductsSearchBar = options.Class.extend({
         }).open();
     },
 });
+
+editorMenu.include({
+    custom_events: _.extend({}, editorMenu.prototype.custom_events || {}, {
+        shop_grid_update: '_onShopGridUpdate'
+    }),
+
+    //--------------------------------------------------------------------------
+    // Public
+    //--------------------------------------------------------------------------
+
+    /**
+     * @override
+     */
+    cancel: function (reload) {
+        return this._super(this.productGridDirty || reload);
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * @private
+     */
+    _getShopContext: function () {
+        var $editable = this.wysiwyg.getEditable();
+        var productGrid = $editable.find('#product_table');
+        return _.pick(productGrid.data(), function (value) {
+            return _.isNumber(value) || _.isString(value);
+        });
+    },
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    /**
+     *
+     * @private
+     */
+    _onShopGridUpdate: function () {
+        var self = this;
+        var $editable = this.wysiwyg.getEditable();
+        var $productGrid = $editable.find('#product_table');
+        this.productGridDirty = true;
+        return this._rpc({
+            route: '/shop/render_grid',
```

Comment:
This should not be the responsability of the website edit menu. Why is not this in the snippet option directly ?